plugins/poll/assets/javascripts/discourse/widgets/discourse-poll.js

```diff
@@ -1,8 +1,20 @@
 import { hbs } from "ember-cli-htmlbars";
```

Comment:
Sorry @merefield we are trying to move away from widgets, I can't accept a PR with 400 new lines of widget code. Maybe we try to extract smaller PRs out of this first please, I would really want to know why you think you need widgets here too.

Reply:
In theory there shouldn't be any need to use widgets at all. You can render a glimmer template directly using the `helper.renderGlimmer` function which is passed to `decorateCookedElement` callbacks.

e.g. https://github.com/discourse/discourse-policy/blob/3547813fe0cf25b53b7ef035f5ca6b207767862f/assets/javascripts/initializers/extend-for-policy.js#L50-L75

If something isn't working with that kind of no-widget strategy, then let's figure it out and fix it.

Reply:
Avatar fix broken out here: #28121 28121

Reply:
> In theory there shouldn't be any need to use widgets at all. You can render a glimmer template directly using the `helper.renderGlimmer` function which is passed to `decorateCookedElement` callbacks.

This is example what is already happening (prior to this PR).  There is nothing additional being rendered by widget code here, it's purely shifting the state management up to the widget level and changing the interface to the Glimmer Component so it is more granular and explicit.